client-runtime/src/main/java/com/microsoft/rest/v2/http/FlowableHttpRequestBody.java

```diff
@@ -42,13 +43,16 @@ public String contentType() {
     }
 
     @Override
-    public InputStream createInputStream() {
+    public Flowable<byte[]> content() {
         return content;
     }
 
     @Override
     public HttpRequestBody buffer() throws IOException {
-        final byte[] bytes = ByteStreams.toByteArray(content);
-        return new ByteArrayHttpRequestBody(bytes, contentType);
+        if (!isBuffered) {
+            content = content.replay().autoConnect();
```

Comment:
Would be good to have automated test coverage for this. Logging is using it and it's working there but you have to specify `LogLevel.BODY`.

Reply:
Modified a RestProxy test to add this RequestPolicy in such a way that `.replay` is required for the test to complete successfully.